lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -388,19 +388,41 @@ xilinx::AIE::HasValidBDs<ConcreteType>::verifyTrait(Operation *op) {
   return success();
 }
 
+// Check that the given DMA-like op (e.g. MemOp, ShimDMAOp)
+// has valid DMA channels.
+template <typename ConcreteType>
+LogicalResult
+xilinx::AIE::HasValidDMAChannels<ConcreteType>::verifyTrait(Operation *op) {
+  auto element = cast<ConcreteType>(op);
+  DenseSet<xilinx::AIE::DMAChannel> used_channels;
+  for (auto &bodyOp : element.getBody().getOps()) {
+    // check for duplicate DMA channels within the same MemTileDMAOp
+    if (auto DMA_start = dyn_cast<xilinx::AIE::DMAStartOp>(bodyOp)) {
+      xilinx::AIE::DMAChannel dmaChan = std::make_pair(
+          DMA_start.getChannelDir(), DMA_start.getChannelIndex());
```

Comment:
I guess that starting with C++17 most of utility factory functions are useless because of CTAD https://en.cppreference.com/w/cpp/language/class_template_argument_deduction
```suggestion
      xilinx::AIE::DMAChannel dmaChan = { DMA_start.getChannelDir(), DMA_start.getChannelIndex() };
```